**Python Problem Solving**


---


7. How to read a number (n) and compute n + nn + nnn in python

In [10]:
a = int(input("Input an integer: "))
n1 = int(" %s" % a)
n2 = int("%s%s" % (a, a))
n3 = int("%s%s%s" % (a, a, a))

print(n1 + n2 + n3)

Input an integer:  5


615


# Explaination:

1. a = int(input("Input an integer: "))
This line takes an integer input from the user and stores it in variable a.

For example, if the user inputs 5, then a =

2. n1 = int(" %s" % a)
This line is a bit unusual due to the extra space. " %s" means format a as a string, but it puts a space before the number.

So, " %s" % a becomes " 5" (a string with a space before 5).

Then it's converted back to an integer using int(), and Python will ignore the space: int(" 5") becomes 5.

Thus, 

3. n2 = int("%s%s" % (a, a))
Here, "%s%s" % (a, a) combines a and a as strings.

If a = 5, then this becomes "55" as a string.

int("55") converts the string to an integer, so n2 

4. n3 = int("%s%s%s" % (a, a, a))
Similarly, this combines three copies of a as a string.

If a = 5, then it becomes "555", and n3 = 5

5. print(n1 + n2 + n3)
Adds all three integers: 5 + 55 + 555 = 615, and prints the result..

55.

n1 = 5. 5.